components/api/src/main/java/com/hotels/styx/api/extension/service/ConnectionPoolSettings.java

```diff
@@ -74,12 +71,12 @@ public ConnectionPoolSettings(int maxConnectionsPerHost,
                                   int socketTimeoutMillis,
                                   int pendingConnectionTimeoutMillis,
                                   long connectionExpirationSeconds) {
-        this.maxConnectionsPerHost = firstNonNull(maxConnectionsPerHost, DEFAULT_MAX_CONNECTIONS_PER_HOST);
-        this.maxPendingConnectionsPerHost = firstNonNull(maxPendingConnectionsPerHost, DEFAULT_MAX_PENDING_CONNECTIONS_PER_HOST);
-        this.connectTimeoutMillis = firstNonNull(connectTimeoutMillis, DEFAULT_CONNECT_TIMEOUT_MILLIS);
-        this.socketTimeoutMillis = firstNonNull(socketTimeoutMillis, DEFAULT_SOCKET_TIMEOUT_MILLIS);
-        this.pendingConnectionTimeoutMillis = firstNonNull(pendingConnectionTimeoutMillis, DEFAULT_CONNECT_TIMEOUT_MILLIS);
-        this.connectionExpirationSeconds = firstNonNull(connectionExpirationSeconds, DEFAULT_CONNECTION_EXPIRATION_SECONDS);
+        this.maxConnectionsPerHost = maxConnectionsPerHost;
+        this.maxPendingConnectionsPerHost = maxPendingConnectionsPerHost;
+        this.connectTimeoutMillis = connectTimeoutMillis;
+        this.socketTimeoutMillis = socketTimeoutMillis;
+        this.pendingConnectionTimeoutMillis = pendingConnectionTimeoutMillis;
+        this.connectionExpirationSeconds = connectionExpirationSeconds;
```

Comment:
This constructor is marked as deprecated. Perhaps this is an opportunity to remove it? AFAICS it is only called from within this file.